pkg/plugins/manager/client/client.go

```diff
@@ -165,3 +182,18 @@ func (s *Service) plugin(ctx context.Context, pluginID string) (*plugins.Plugin,
 
 	return p, true
 }
+
+func (s *Service) attachJWT(ctx context.Context, pluginCtx backend.PluginContext) context.Context {
+	if pluginCtx.User == nil {
+		return ctx
+	}
+
+	token, err := s.jwtAuthService.Generate(pluginCtx.User.Login, pluginCtx.PluginID)
+	if err != nil {
+		return ctx
```

Comment:
If this is a real error and not "JWT token genertion is disabled" we swallow/disregard it. That is not good in general. Suggest checking if the `jwtAuthService.IsEnabled` or similar before calling `Generate` and in case of error at least log the error if you can't return it from here.

Reply:
good call, i'll handle this in https://github.com/grafana/grafana/pull/57420. marking it as resolved here since it's no longer part of this PR